pkg/config/schema/resource/schema.go

```diff
@@ -99,6 +100,9 @@ type Builder struct {
 	// Proto refers to the protobuf message type name corresponding to the type
 	Proto string
 
+	// ReflectType is the type of the go struct
+	ReflectType reflect.Type
```

Comment:
Maybe update the comments here and for `Proto`/`ProtoPackage` to make it clear how they work together.